### PR TITLE
feat: Support for Windows.Data.Xml.Dom APIs

### LINKED ISSUE #3069: Support for Windows.Data.Xml.Dom
## What would you like to be added:

Support for `Windows.Data.Xml.Dom`.

## Why is this needed:

Although matching functionality is in `System.Xml`, this specific API is required for proper support for badge notifications/toast notifications. It is quite easy to just forward the API methods to `System.Xml`

## For which Platform:

- [x] iOS
- [x] Android
- [x] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [x] macOS
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #3069 

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

APIs not supported.

## What is the new behavior?

All APIs which can be forwarded to `System.Xml` are implemented.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

**Comment by MartinZikmund** [comment]
Will add some simple unit tests.

File: src/Uno.UI.Tests/Windows_Data_Xml/DomTests/Given_XmlDocument.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+			var badgeAttributes = badgeXml.GetElementsByTagName("badge");
+			((XmlElement)badgeAttributes[0]).SetAttribute("value", "26");
+			var xml = badgeXml.GetXml();
+			Assert.AreEqual("<badge value=\"26\" />", xml);
+		}
+
+		private string GetTestXml()
+		{
+			using var stream = typeof(Given_XmlDocument).Assembly.GetManifestResourceStream("Uno.UI.Tests.Windows_Data_Xml.DomTests.basictest.xml");
+			using var streamReader = new StreamReader(stream);
```
[the remark this answers is not in the document]
Comment (MartinZikmund) [reply src/Uno.UI.Tests/Windows_Data_Xml/DomTests/Given_XmlDocument.cs:64]: Not applicable

**Comment by MartinZikmund** [comment]
@jeromelaban Should I create a new section in PackageDiffIgnore? What would be the base version for it?

**Comment by jeromelaban** [comment]
Yes you can. Use a section similar to this one:

https://github.com/unoplatform/uno/pull/3044/files#diff-49d44d5b2511b8a0bd889bc14a811169R809

**Comment by jeromelaban** [comment]
Thanks Martin!

File: src/Uno.UI.Tests/Windows_Data_Xml/DomTests/Given_XmlDocument.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (MartinZikmund) [reply src/Uno.UI.Tests/Windows_Data_Xml/DomTests/Given_XmlDocument.cs:64]: Not applicable

**Comment by pkar70** [comment]
I will use it in my Live Tile implementation :)

File: src/Uno.UI.Tests/Windows_Data_Xml/DomTests/Given_XmlDocument.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (MartinZikmund) [reply src/Uno.UI.Tests/Windows_Data_Xml/DomTests/Given_XmlDocument.cs:64]: Not applicable

**Pull request merged**